bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductteaserImpl.java

```diff
@@ -0,0 +1,147 @@
+
+    protected static final String RESOURCE_TYPE = "venia/components/commerce/product/v1/Productteaser";
+    private static final Logger LOG = LoggerFactory.getLogger(ProductteaserImpl.class);
+    private static final String PRODUCT_PATH_PROP = "productSku";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @Inject
+    private Resource resource;
+
+    @Inject
+    private Page currentPage;
+
+    private ProductInterface product;
+
+
+    private NumberFormat priceFormatter;
+
+
+    @Inject
+    private ProductGraphqlService productGraphqlService;
+
+
+    @ScriptVariable
+    private ValueMap properties;
+
+    private Page productPage;
+
+    @PostConstruct
+    private void initModel() {
+        String sku = null;
+        String productPath = null;
+        productPage = Utils.getProductPage(currentPage);
+        if (productPage == null) {
+            productPage = currentPage;
+        }
+        productPath = properties.get(PRODUCT_PATH_PROP, String.class);
+        if (productPath != null && !productPath.isEmpty()) {
+            sku = getSkuFromUrl(productPath);
+            GraphqlClient client = resource.adaptTo(GraphqlClient.class);
+            if (client == null) {
+                LOG.error("Cannot get a GraphqlClient using the resource at {}", resource.getPath());
+            }
+            Locale locale = currentPage.getLanguage(false);
+            this.product = productGraphqlService.getProduct(client, sku);
+            this.priceFormatter = Utils.buildPriceFormatter(locale, this.getCurrency());
+        }
+    }
+
+    @Override
+    public String getName() {
+        return (this.product != null ? this.product.getName() : null);
+
+    }
+
+    @Override
+    public String getFormattedPrice() {
+
+        Double price = this.getPrice();
+
+        if (price != null) {
+            return this.priceFormatter.format(price);
+        }
+        return null;
+    }
+
+    @Override
+    public String getUrl() {
+        return (this.product != null ? String.format("%s.%s.html", productPage.getPath(), product.getUrlKey()) : null);
+    }
+
+    @Override
+    public List<Asset> getAssets() {
+        if (this.product != null) {
+            return productGraphqlService.filterAndSortAssets(this.product.getMediaGalleryEntries());
+        }
+        return null;
+    }
+
+    private String getCurrency() {
+        if (this.product != null) {
+            return this.product.getPrice().getRegularPrice().getAmount().getCurrency().toString();
+        }
+        return null;
+    }
+
+    private Double getPrice() {
+        if (this.product != null) {
+            return this.product.getPrice().getRegularPrice().getAmount().getValue();
+        }
+        return null;
+    }
+
+    // Product DnD from content finder  provides only path of the product to be replaced in
+    private String getSkuFromUrl(String url) {
+        if (url != null && !url.isEmpty() && url.contains("/")) {
```

Comment:
Better use `StringUtils.substringAfterLast`, make it easier and you don't have to check for `null`.